### PR TITLE
[full-ci] style file versions list

### PR DESCRIPTION
**Pull request by hurradieweltgehtunter**
## Description
Beautify file versions list

## Related Issue
- Fixes https://github.com/owncloud/web/issues/8503

## Motivation and Context
Before:
grafik

After:
grafik

## Types of changes
- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Technical debt
- [ ] Tests

## Checklist:
- [ ] Code changes
- [ ] Unit tests added
- [ ] Acceptance tests added
- [x] Documentation ticket raised: https://github.com/owncloud/docs-webui/issues/118

## Open tasks:
- [ ] Check and possibly fix tests

### CODE REVIEW

**Comment by hurradieweltgehtunter** [comment]
Need help with some topics. 
- See 2 comments in css section
- tests: 
grafik

**Comment by hurradieweltgehtunter** [comment]
If the height (= length) of the list becomes an issue, a customized — but not favored — version would look like:

grafik

**Comment by JammingBen** [comment]
Love the design, especially the "timeline"-feeling because of the continuous line on the left side!

The actions take a lot of vertical space though. Maybe keep the actions in the right side? Like this:

image

**Comment by hurradieweltgehtunter** [comment]
> Love the design, especially the "timeline"-feeling because of the continuous line on the left side!
> 
> The actions take a lot of vertical space though. Maybe keep the actions in the right side? Like this:
> 
> image

Idea was, to have it aligned with other action buttons in the sidebar. IMO the length of the list isn't problematic. Also, showing only icons isn't very intuitive. Text-based buttons work much better regarding accessibility.

**Comment by hurradieweltgehtunter** [comment]
Added negative margin css classes to design system. Ok, to have it in the same PR?

**Comment by JanAckermann** [comment]
> Added negative margin css classes to design system. Ok, to have it in the same PR? 

Totally fine, but you want to add a changelog in the odd/changelog dir

**Comment by hurradieweltgehtunter** [comment]
> > Added negative margin css classes to design system. Ok, to have it in the same PR?
> 
> Totally fine, but you want to add a changelog in the odd/changelog dir

Changelog file is included. Make 2, one for version list styling, one for design system?

**Comment by JanAckermann** [comment]
> > > Added negative margin css classes to design system. Ok, to have it in the same PR?
> 
> > 
> 
> > Totally fine, but you want to add a changelog in the odd/changelog dir
> 
> 
> 
> Changelog file is included. Make 2, one for version list styling, one for design system?

Yes

**Comment by tbsbdr** [comment]
timeline style - awesome 👍

**Comment by JammingBen** [comment]
@hurradieweltgehtunter I fixed the linter and removed the failing unit test (which is not needed anymore due to the removal of the resource icons). Let's see how the pipeline does.

**Comment by ownclouders** [comment]
Results for <strong>acceptance oCIS</strong> https://drone.owncloud.com/owncloud/web/33156/61/1
<details><summary>:boom: The acceptance tests failed on retry. Please find the screenshots inside ...</summary>

<p>


### webUIFilesActionMenu-versions_feature-L15.png
  

### webUIFilesActionMenu-versions_feature-L26.png
  


</p></details>
:boom: The oCISFiles5 tests pipeline failed. The build has been cancelled.

**Comment by JammingBen** [comment]
Okay, so it seems that we have to adjust some selectors for the acceptance tests as well. I can have a look after the meetings later, shouldn't be too complicated.

File: packages/web-app-files/src/components/SideBar/Versions/FileVersions.vue [hunk 1]
```diff
@@ -1,47 +1,50 @@
...
-            data-testid="file-versions-file-last-modified-date"
-          >
-            {{ formatVersionDate(item) }}
-          </oc-td>
-          <oc-td class="oc-text-muted oc-text-nowrap" data-testid="file-versions-file-size">
-            {{ formatVersionFileSize(item) }}
-          </oc-td>
-          <oc-td v-if="isRevertable" width="shrink">
+    <ul v-if="!loading && hasVersion" class="oc-m-rm oc-position-relative">
+      <li class="spacer oc-pb-l"></li>
```
Comment (kulmann) [inline packages/web-app-files/src/components/SideBar/Versions/FileVersions.vue:5]: Please add `aria-hidden="true"`. This element is only for styling, probably confusing for screen readers to have an empty list item.

File: packages/web-app-files/src/components/SideBar/Versions/FileVersions.vue [hunk 2]
```diff
@@ -139,3 +148,56 @@ export default defineComponent({
...
+        height: 11px;
+        position: absolute;
+        left: -6px;
+        top: 4px;
+        background-color: var(--oc-color-border);
+        border-radius: 50%;
+      }
+      .version-details {
+        .version-date {
+          font-weight: 600; // No css class for font-weight?
```
Comment (kulmann) [inline packages/web-app-files/src/components/SideBar/Versions/FileVersions.vue:182]: feel free to add it in `packages/design-system/src/tokens/ods/font.yaml`. :-) You could add a `weight` section and then add some reasonable values.

**Review by kulmann — CHANGES_REQUESTED** [review]

File: packages/web-app-files/src/components/SideBar/Versions/FileVersions.vue (hunk 2 above)
Comment (kulmann) [reply packages/web-app-files/src/components/SideBar/Versions/FileVersions.vue:182]: if you don't want to add the weight var, just delete the comment, fine as well

File: packages/web-app-files/src/components/SideBar/Versions/FileVersions.vue (hunk 1 above)
Comment (hurradieweltgehtunter) [reply packages/web-app-files/src/components/SideBar/Versions/FileVersions.vue]: done

File: packages/web-app-files/src/components/SideBar/Versions/FileVersions.vue (hunk 2 above)
Comment (hurradieweltgehtunter) [reply packages/web-app-files/src/components/SideBar/Versions/FileVersions.vue]: ok added. how do I compile the yml file?

Comment (kulmann) [reply packages/web-app-files/src/components/SideBar/Versions/FileVersions.vue]: pnpm build:tokens

that’s also part of pnpm build

Comment (hurradieweltgehtunter) [reply packages/web-app-files/src/components/SideBar/Versions/FileVersions.vue]: done.
How would one now programmatically create corresponding css classes? Like for spacing:
```
@each $sizeKey, $sizeValue in $sizes {
  &-#{$sizeKey} {
    #{$directiveValue}-#{$direction}: $sizeValue !important;
  }
}
```

File: changelog/unreleased/enhancement-add-negative-margin-design-system [hunk 3]
```diff
@@ -0,0 +1,7 @@
+Enhancement: Design Sytem: CSS classes for negative margin added
```
[the remark this answers is not in the document]
Comment (hurradieweltgehtunter) [reply changelog/unreleased/enhancement-add-negative-margin-design-system]: ah ok. sorry, didnt knew.
done

File: changelog/unreleased/change-re-style-file-versions [hunk 4]
```diff
@@ -0,0 +1,7 @@
+Change: Beautify file version list
```
[the remark this answers is not in the document]
Comment (hurradieweltgehtunter) [reply changelog/unreleased/change-re-style-file-versions]: done

**Comment by hurradieweltgehtunter** [comment]
@JammingBen PR is done from my side, but I need help with the test suite. Could you fix the tests?

**Comment by JammingBen** [comment]
> @JammingBen PR is done from my side, but I need help with the test suite. Could you fix the tests?

Looks good now, no?

**Pull request merged**